examples/language_model/gpt-3/dygraph/dataset.py

```diff
@@ -0,0 +1,390 @@
+from paddlenlp.data import Stack, Tuple, Pad
+from paddlenlp.utils.log import logger
+
+
+def construct_samples_and_shuffle_data(name, data_prefix, documents, sizes,
+                                       num_samples, seq_length, seed,
+                                       build_data_file):
+    """
+    documents: document index from 0 to len(docs)
+    sizes: the length list of all docs.
+    num_samples: total step*bs iterations of data.
+    seq_length: the sequence length.
+
+
+    sum(sizes) = tokens_per_epoch
+    data_nums = num_samples *  local_batch_size
+    num_epochs = (data_nums + 1) // sum(sizes)
+    len(doc_idx) = num_epochs * sum(sizes)
+
+    """
+    # Number of tokens in each epoch and number of required epochs.
+    tokens_per_epoch = _num_tokens(documents, sizes)
+    num_epochs = _num_epochs(tokens_per_epoch, seq_length, num_samples)
+    # Rng state
+    np_rng = np.random.RandomState(seed=seed)
+
+    # Filename of the index mappings.
+    _filename = data_prefix
+    _filename += '_{}_indexmap'.format(name)
+    _filename += '_{}ns'.format(num_samples)
+    _filename += '_{}sl'.format(seq_length)
+    doc_idx_filename = _filename + '_doc_idx.npy'
+    sample_idx_filename = _filename + '_sample_idx.npy'
+    shuffle_idx_filename = _filename + '_shuffle_idx.npy'
+
+    # support multi-machines
+    if (not os.path.isfile(doc_idx_filename)) or \
+        (not os.path.isfile(sample_idx_filename)) or \
+        (not os.path.isfile(shuffle_idx_filename)):
+        if num_epochs == 1:
+            separate_last_epoch = False
+        else:
+            num_samples_from_epochs_minus_one = (
+                (num_epochs - 1) * tokens_per_epoch - 1) // seq_length
+            last_epoch_num_samples = num_samples - \
+                                        num_samples_from_epochs_minus_one
+            assert last_epoch_num_samples >= 0, \
+                'last epoch number of samples should be non-negative.'
+            num_samples_per_epoch = (tokens_per_epoch - 1) // seq_length
+            assert last_epoch_num_samples < (num_samples_per_epoch + 1), \
+                'last epoch number of samples exceeded max value.'
+            separate_last_epoch = (
+                last_epoch_num_samples < int(0.80 * num_samples_per_epoch))
+        doc_idx = _build_doc_idx(documents, num_epochs, np_rng,
+                                 separate_last_epoch)
+        # sample-idx.
+        assert doc_idx.dtype == np.int32
+        sample_idx = _build_sample_idx(sizes, doc_idx, seq_length, num_epochs,
+                                       tokens_per_epoch)
+        if separate_last_epoch:
+            num_samples_ = num_samples_from_epochs_minus_one
+        else:
+            num_samples_ = sample_idx.shape[0] - 1
+        shuffle_idx = _build_shuffle_idx(num_samples_, sample_idx.shape[0] - 1,
+                                         np_rng)
+        if paddle.distributed.get_rank() % 8 == 0:
```

Comment:
`8` for hard code here.  you can pass local_rank as  https://github.com/PaddlePaddle/PaddleNLP/pull/930

Reply:
ok， fix it